### ISSUE TITLE
TRACEPOINT_PROBE can only get one frame of call stack

### ISSUE THREAD

**Issue by adam900710** [iovisor/bcc#2332]
I tried the following two different scripts to get call stack.
The following one using TRACEPOINT_PROBE can only get one frame of call stack:
```
#!/usr/bin/python2
# @lint-avoid-python-3-compatibility-imports
#
from __future__ import print_function
from bcc import BPF
from sys import stderr
from sys import argv

text_bpf = '''
#include <uapi/linux/ptrace.h>
#define FSID_SIZE   16
struct data_t {
    u64 stack_id;
};

BPF_PERF_OUTPUT(events);
BPF_STACK_TRACE(stack_traces, 128);

TRACEPOINT_PROBE(btrfs, btrfs_reserve_extent)
{
    u64 pid = bpf_get_current_pid_tgid();
    struct data_t data;

    data.stack_id = stack_traces.get_stackid(args, BPF_F_REUSE_STACKID);

    events.perf_submit(args, &data ,sizeof(data));
    return 0;
}
'''

def process_event(cpu, data, size):
    event = b["events"].event(data)
    for stack in stack_traces.walk(event.stack_id):
        sym = b.ksym(stack)
        print(sym)
    print()


b = BPF(text = text_bpf)

b["events"].open_perf_buffer(process_event, page_cnt=64)
stack_traces = b.get_table("stack_traces")
print("start recording", file=stderr)
while 1:
    try:
        b.perf_buffer_poll()
    except:
        exit(0)
```
The result is:
```
start recording
find_free_extent

...
```

While using kprobe attach, like the following script, everything works as expected:
```
#!/usr/bin/python2
# @lint-avoid-python-3-compatibility-imports
#
from __future__ import print_function
from bcc import BPF
from sys import stderr

text_bpf = '''
#include <uapi/linux/ptrace.h>
#define FSID_SIZE   16
struct data_t {
    u64 stack_id;
};

BPF_PERF_OUTPUT(events);
BPF_STACK_TRACE(stack_traces, 128);

int trace_stack(struct pt_regs *ctx)
{
    u64 pid = bpf_get_current_pid_tgid();
    struct data_t data;

    data.stack_id = stack_traces.get_stackid(ctx, BPF_F_REUSE_STACKID);

    events.perf_submit(ctx, &data ,sizeof(data));
    return 0;
}
'''

def process_event(cpu, data, size):
    event = b["events"].event(data)
    for stack in stack_traces.walk(event.stack_id):
        sym = b.ksym(stack)
        print(sym)
    print()


b = BPF(text = text_bpf)

b["events"].open_perf_buffer(process_event, page_cnt=64)
b.attach_kprobe(event="submit_one_bio", fn_name="trace_stack")
stack_traces = b.get_table("stack_traces")
print("start recording", file=stderr)
while 1:
    try:
        b.perf_buffer_poll()
    except:
        exit(0)
```

The result is the full stack:
```
btrfs_reserve_extent
btrfs_alloc_tree_block
alloc_tree_block_no_bg_flush
__btrfs_cow_block
btrfs_cow_block
btrfs_search_slot
btrfs_del_csums
...
```

Any clue why this happens?

**Comment by yonghong-song** [comment]
The tracepoint issue is a known issue and will be fixed in 5.2.
https://lkml.org/lkml/2019/4/29/67

You kernel config mostly have
```
CONFIG_UNWINDER_ORC=y
```
I guess it will be backported to some old kernels (esp. stable ones) as well.

**Comment by adam900710** [comment]
Thank you!

After switching to frame pointer unwinder, even without that patch it works fine now.
